File: problem_2/problem_2.go
package main

import "fmt"

func getEvenNumbers(numbers []int) []int {
    evens := []int{}
    for _, n := range numbers {
        if n%2 == 0 {
            evens = append(evens, n)
        }
    }
    return evens
}

func main() {
    numbers := []int{1, 2, 3, 4, 5}
    fmt.Println(getEvenNumbers(numbers))
}

File: problem_2/problem_2.rs
fn get_even_numbers(numbers: &[i32]) -> Vec<i32> {
    numbers.iter().filter(|&n| n % 2 == 0).cloned().collect()
}

fn main() {
    let numbers = vec![1, 2, 3, 4, 5];
    println!("{:?}", get_even_numbers(&numbers));
}
